### PR TITLE
Io #27

### PR DESCRIPTION
**Pull request by neutronest**
change message:

support read from ND4J INDArray

support read from csv

### CODE REVIEW

File: src/main/scala/com/framex/core/ElemX.scala [hunk 1]
```diff
@@ -61,6 +60,14 @@ object ElemX {
     }
   }
 
+  def castToDouble(data: String) : ElemX = {
+    try {
+      ElemX(ExDouble(data.toDouble))
+    } catch {
+      case _ => ElemX(ExString(data))
```
Comment (izhangzhihao) [inline src/main/scala/com/framex/core/ElemX.scala]: Why `castToDouble` return `ElemX(ExString(.))` when error happend?

Comment (neutronest) [reply src/main/scala/com/framex/core/ElemX.scala]: It's now used in reading csv.

When you read data from csv, the default ElemX for each element is ExString. It's used to "check if one element can be cast to numeric or not, if success, then cast to double, or just use default string type".

Comment (neutronest) [reply src/main/scala/com/framex/core/ElemX.scala]: Maybe we need a good name for this method, or refactor it's logic...

Comment (izhangzhihao) [reply src/main/scala/com/framex/core/ElemX.scala]: What if we let the user decide the element type...

Comment (izhangzhihao) [reply src/main/scala/com/framex/core/ElemX.scala]: Like `readCsv[Int,Double,String]("/path")` or something else...

Comment (neutronest) [reply src/main/scala/com/framex/core/ElemX.scala]: Let' me think a few well...

**Comment by izhangzhihao** [comment]
What if the csv file is 

```
PassengerId,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
892.1,3,"Kelly, Mr. James",male,34.5,0,0,330911,7.8292,,Q
"893",3,"Wilkes, Mrs. James (Ellen Needs)",female,47,1,0,363272,7,,S
```

In this situation, should we throw this error early, to say that "the two elements' type does not match ", or throw an exception that the column contains a different type of elements?

**Comment by neutronest** [comment]
I don't understand the first one... Second one is ok for this situation I think. Thanks for the advice! Actually more work need to be done to check the type of original data from csv file.

**Comment by neutronest** [comment]
Now the readCSV method will check each column's data that can convert to double or not. Only if all the data in one column can convert to double without any exception, otherwise the type is string by default.

**Comment by neutronest** [comment]
@izhangzhihao check..check..

**Pull request merged**